### PR TITLE
docker: support for testing snapcraft in proposed

### PR DESCRIPTION
**Pull request by sergiusens**
Adding a dockerfile specifically tailored for testing the snapcraft deb
in xenial-proposed.

Also remove the multiverse sed from the standard Dockerfile as
multiverse is already in the base image.

LP: #1791116
Signed-off-by: Sergio Schvezov <sergio.schvezov@canonical.com>

- [x] Have you followed the [guidelines for contributing](https://github.com/snapcore/snapcraft/blob/master/CONTRIBUTING.md)?
- [x] Have you signed the [CLA](http://www.ubuntu.com/legal/contributors/)?
- [x] If this is a bugfix. Have you checked that there is a bug report open for the issue you are trying to fix on [bug reports](https://bugs.launchpad.net/snapcraft)?
- [x] If this is a new feature. Have you discussed the design on the [forum](https://forum.snapcraft.io)?
- [x] Have you successfully run `./runtests.sh static`?
- [x] Have you successfully run `./runtests.sh tests/unit`?

-----

### CODE REVIEW

File: docker/xenial-proposed.Dockerfile [hunk 1]
```diff
@@ -0,0 +1,21 @@
+FROM ubuntu:xenial
+
+# Enable proposed and pin snapcraft
+RUN echo "deb http://archive.ubuntu.com/ubuntu/ xenial-proposed restricted main multiverse universe" >> /etc/apt/sources.list
+RUN cat /etc/apt/sources.list
```
Comment (kyrofa) [inline docker/xenial-proposed.Dockerfile]: Is this needed in the final version?

File: docker/xenial-proposed.Dockerfile [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+  apt-get dist-upgrade --yes && \
+  apt-get install --yes \
+  git \
+  snapcraft/xenial-proposed \
+  && \
+  apt-get autoclean --yes && \
+  apt-get clean --yes
+
+# Required by click.
+ENV LC_ALL C.UTF-8
```
Comment (kyrofa) [inline docker/xenial-proposed.Dockerfile]: This is no longer technically necessary.

File: docker/xenial-proposed.Dockerfile (hunk 1 above)
Comment (sergiusens) [reply docker/xenial-proposed.Dockerfile]: heh, not necessarily, I will remove

File: docker/xenial-proposed.Dockerfile (hunk 2 above)
Comment (sergiusens) [reply docker/xenial-proposed.Dockerfile]: fi not set, you get a warning, so I'd say we leave it as we do not silently set it.

**Comment by sergiusens** [comment]
Also, 
```
multipass@https-test:~/t$ sudo docker run -it snapcore/snapcraft:latest-proposed  bash
root@b85308a177fd:/# mkdir t
root@b85308a177fd:/# cd t
root@b85308a177fd:/t# ls
root@b85308a177fd:/t# snapcraft init
Created snap/snapcraft.yaml.
Edit the file to your liking or run `snapcraft` to get started
root@b85308a177fd:/t# snapcraft
Pulling my-part 
Building my-part 
Staging my-part 
Priming my-part 
Snapping 'my-snap-name' |                                                                         
Snapped my-snap-name_0.1_amd64.snap
```

and
```
root@e4fe59e10349:/# mv /usr/bin/mksquashfs /usr/bin/mksquashfs.bak
root@e4fe59e10349:/# mkdir t
root@e4fe59e10349:/# cd t
root@e4fe59e10349:/t# snapcraft init
Created snap/snapcraft.yaml.
Edit the file to your liking or run `snapcraft` to get started
root@e4fe59e10349:/t# snapcraft
Pulling my-part 
Building my-part 
Staging my-part 
Priming my-part 
A tool snapcraft depends on could not be found: 'mksquashfs'.
Ensure the tool is installed and available, and try again.
Sorry, an error occurred in Snapcraft:
Traceback (most recent call last):
  File "/usr/bin/snapcraft", line 9, in <module>
    load_entry_point('snapcraft==2.43.1', 'console_scripts', 'snapcraft')()
  File "/usr/lib/python3/dist-packages/click/core.py", line 716, in __call__
    return self.main(*args, **kwargs)
  File "/usr/lib/python3/dist-packages/click/core.py", line 696, in main
    rv = self.invoke(ctx)
  File "/usr/lib/python3/dist-packages/click/core.py", line 1037, in invoke
    return Command.invoke(self, ctx)
  File "/usr/lib/python3/dist-packages/click/core.py", line 889, in invoke
    return ctx.invoke(self.callback, **ctx.params)
  File "/usr/lib/python3/dist-packages/click/core.py", line 534, in invoke
    return callback(*args, **kwargs)
  File "/usr/lib/python3/dist-packages/click/decorators.py", line 17, in new_func
    return f(get_current_context(), *args, **kwargs)
  File "/usr/lib/python3/dist-packages/snapcraft/cli/_runner.py", line 93, in run
    ctx.forward(lifecyclecli.commands["snap"])
  File "/usr/lib/python3/dist-packages/click/core.py", line 552, in forward
    return self.invoke(cmd, **kwargs)
  File "/usr/lib/python3/dist-packages/click/core.py", line 534, in invoke
    return callback(*args, **kwargs)
  File "/usr/lib/python3/dist-packages/snapcraft/cli/lifecycle.py", line 139, in snap
    snap_name = lifecycle.pack(directory, output)
  File "/usr/lib/python3/dist-packages/snapcraft/internal/lifecycle/_packer.py", line 45, in pack
    mksquashfs_path = file_utils.get_tool_path("mksquashfs")
  File "/usr/lib/python3/dist-packages/snapcraft/file_utils.py", line 355, in get_tool_path
    raise ToolMissingError(command_name=command_name)
snapcraft.internal.errors.ToolMissingError: A tool snapcraft depends on could not be found: 'mksquashfs'.
Ensure the tool is installed and available, and try again.
We would appreciate it if you created a bug report at
https://launchpad.net/snapcraft/+filebug with the above text included.
```

**Pull request merged**